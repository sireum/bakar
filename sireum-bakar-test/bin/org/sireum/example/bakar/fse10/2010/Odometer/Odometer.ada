package Odometer
  --# own Trip, Total : Integer;
is
   procedure Zero_Trip;
   --# global out Trip;
   --# derives Trip from;
   --# post Trip = 0;

   function Read_Trip return Integer;
   --# global in Trip;
   --# return Trip;

   function Read_Total return Integer;
   --# global in Total;
   --# return Total;

   procedure Inc;
   --# global in out Trip, Total;
   --# derives Trip from Trip & Total from Total;
   --# pre  Trip + 1 < Integer'Last and
   --#      Total + 1 < Integer'Last;
   --# post Trip = Trip~ + 1 and
   --#      Total = Total~ + 1;

private

   procedure Set_Trip(T : Integer);
   --# global out Trip;
   --# derives Trip from T;
   --# post Trip = T;

   procedure Set_Total(T : Integer);
   --# global out Total;
   --# derives Total from T;
   --# post Total = T;

end Odometer;

package body Odometer is
   Trip : Integer;
   Total : Integer;

   procedure Zero_Trip is
   begin
      Trip := 0;
   end Zero_Trip;

   function Read_Trip return Integer is
   begin
      return Trip;
   end Read_Trip;

   function Read_Total return Integer is
   begin
      return Total;
   end Read_Total;

   procedure Inc is
   begin
      Trip := Trip + 1;
      Total := Total + 1;
   end Inc;

   procedure Set_Trip (T: Integer) is
   begin
      Trip := T;
   end Set_Trip;

   procedure Set_Total (T: Integer) is
   begin
      Total := T;
   end Set_Total;

end Odometer;
