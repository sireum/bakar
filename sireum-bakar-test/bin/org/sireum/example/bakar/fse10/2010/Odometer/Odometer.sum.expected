-------------------------------------------------------------------------------
                          Semantic Analysis Summary                            
                                POGS GPL 2010                                  
            Copyright (C) 2010 Altran Praxis Limited, Bath, U.K.               
-------------------------------------------------------------------------------

Summary of:

Verification Condition files (.vcg)
Simplified Verification Condition files (.siv)
Proof Logs (.plg)
Dead Path Conjecture files (.dpc)
Summary Dead Path files (.sdp)

"status" column keys:
    1st character:
        '-' - No VC
        'S' - No SIV
        'U' - Undischarged
        'E' - Proved by Examiner
        'I' - Proved by Simplifier by Inference
        'X' - Proved by Simplifier by Contradiction
        'P' - Proved by Simplifier using User Defined Proof Rules
        'C' - Proved by Checker
        'R' - Proved by Review
        'F' - VC is False
    2nd character:
        '-' - No DPC
        'S' - No SDP
        'U' - Unchecked
        'D' - Dead path
        'L' - Live path

in the directory:
/Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Odometer

Summary produced: 16-SEP-2010 17:29:05.83

File /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Odometer/odometer/inc.vcg
procedure Odometer.Inc

VCs generated 16-SEP-2010 17:29:04

VCs simplified 16-SEP-2010 17:29:04

VCs for procedure_inc :
 -----------------------------------------------------------------------------
| #   | From  | To                  | Proved By          | Dead Path | Status |
|-----------------------------------------------------------------------------
| 1   | start | rtc check @ 60      | Inference          | No DPC    |   I-   |
| 2   | start | rtc check @ 61      | Inference          | No DPC    |   I-   |
| 3   | start |    assert @ finish  | Inference          | No DPC    |   I-   |
 -----------------------------------------------------------------------------


File /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Odometer/odometer/read_total.vcg
function Odometer.Read_Total

VCs generated 16-SEP-2010 17:29:04

VCs simplified 16-SEP-2010 17:29:04

VCs for function_read_total :
 -----------------------------------------------------------------------------
| #   | From  | To                  | Proved By          | Dead Path | Status |
|-----------------------------------------------------------------------------
| 1   | start |    assert @ finish  | Inference          | No DPC    |   I-   |
 -----------------------------------------------------------------------------


File /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Odometer/odometer/read_trip.vcg
function Odometer.Read_Trip

VCs generated 16-SEP-2010 17:29:04

VCs simplified 16-SEP-2010 17:29:04

VCs for function_read_trip :
 -----------------------------------------------------------------------------
| #   | From  | To                  | Proved By          | Dead Path | Status |
|-----------------------------------------------------------------------------
| 1   | start |    assert @ finish  | Inference          | No DPC    |   I-   |
 -----------------------------------------------------------------------------


File /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Odometer/odometer/set_total.vcg
procedure Odometer.Set_Total

VCs generated 16-SEP-2010 17:29:04

VCs simplified 16-SEP-2010 17:29:05

VCs for procedure_set_total :
 -----------------------------------------------------------------------------
| #   | From  | To                  | Proved By          | Dead Path | Status |
|-----------------------------------------------------------------------------
| 1   | start | rtc check @ 71      | Inference          | No DPC    |   I-   |
| 2   | start |    assert @ finish  | Inference          | No DPC    |   I-   |
 -----------------------------------------------------------------------------


File /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Odometer/odometer/set_trip.vcg
procedure Odometer.Set_Trip

VCs generated 16-SEP-2010 17:29:04

VCs simplified 16-SEP-2010 17:29:05

VCs for procedure_set_trip :
 -----------------------------------------------------------------------------
| #   | From  | To                  | Proved By          | Dead Path | Status |
|-----------------------------------------------------------------------------
| 1   | start | rtc check @ 66      | Inference          | No DPC    |   I-   |
| 2   | start |    assert @ finish  | Inference          | No DPC    |   I-   |
 -----------------------------------------------------------------------------


File /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Odometer/odometer/zero_trip.vcg
procedure Odometer.Zero_Trip

VCs generated 16-SEP-2010 17:29:04

VCs simplified 16-SEP-2010 17:29:05

VCs for procedure_zero_trip :
 -----------------------------------------------------------------------------
| #   | From  | To                  | Proved By          | Dead Path | Status |
|-----------------------------------------------------------------------------
| 1   | start | rtc check @ 45      | Inference          | No DPC    |   I-   |
| 2   | start |    assert @ finish  | Inference          | No DPC    |   I-   |
 -----------------------------------------------------------------------------


===============================================================================
Summary:

Proof strategies used by subprograms
-------------------------------------------------------------------------
Total subprograms with at least one VC proved by examiner:              0
Total subprograms with at least one VC proved by simplifier:            6
Total subprograms with at least one VC proved by contradiction:         0
Total subprograms with at least one VC proved with user proof rule:     0
Total subprograms with at least one VC proved using checker:            0
Total subprograms with at least one VC discharged by review:            0

Maximum extent of strategies used for fully proved subprograms:
-------------------------------------------------------------------------
Total subprograms with proof completed by examiner:                     0
Total subprograms with proof completed by simplifier:                   6
Total subprograms with proof completed with user defined rules:         0
Total subprograms with proof completed by checker:                      0
Total subprograms with VCs discharged by review:                        0

Overall subprogram summary:
-------------------------------------------------------------------------
Total subprograms fully proved:                                         6
Total subprograms with at least one undischarged VC:                    0
Total subprograms with at least one false VC:                           0
                                                                    -----
Total subprograms for which VCs have been generated:                    6


ZombieScope Summary:
-------------------------------------------------------------------------
Total subprograms for which DPCs have been generated:                   0
Total number subprograms with dead paths found:                         0
Total number of dead paths found:                                       0


VC summary:
-------------------------------------------------------------------------
Note: U/R denotes where the Simplifier has proved VCs using one or more user-
defined proof rules.

Total VCs by type:                                       
                            -----------Proved By Or Using------------
                     Total  Examiner Simp(U/R)  Checker Review False Undiscgd
Assert or Post:          6      0      6              0      0      0       0
Precondition check:      0      0      0              0      0      0       0
Check statement:         0      0      0              0      0      0       0
Runtime check:           5      0      5              0      0      0       0
Refinement VCs:          0      0      0              0      0      0       0
Inheritance VCs:         0      0      0              0      0      0       0
===============================================================================
Totals:                 11      0     11              0      0      0       0
% Totals:                       0%   100%             0%     0%     0%      0%
===================== End of Semantic Analysis Summary ========================
